File: ScriptService/Dto/Tasks/ScheduledDays.cs
using System;

namespace ScriptService.Dto.Tasks {

    /// <summary>
    /// days when tasks are scheduled
    /// </summary>
    [Flags]
    public enum ScheduledDays {

        /// <summary>
        /// no days selected. Usually leads to all days scheduled except when used as filter
        /// </summary>
        None=0,

        /// <summary>
        /// monday
        /// </summary>
        Monday=1,

        /// <summary>
        /// tuesday
        /// </summary>
        Tuesday=2,

        /// <summary>
        /// wednesday
        /// </summary>
        Wednesday=4,

        /// <summary>
        /// thursday
        /// </summary>
        Thursday=8,

        /// <summary>
        /// friday
        /// </summary>
        Friday=16,

        /// <summary>
        /// saturday
        /// </summary>
        Saturday=32,

        /// <summary>
        /// sunday
        /// </summary>
        Sunday=64,

        /// <summary>
        /// all days selected
        /// </summary>
        All=127
    }
}

File: nc-scripts/src/app/dto/tasks/scheduleddays.ts

/**
 * days when task execution is scheduled
 */
export enum ScheduledDays {

    /**
     * no days selected. Usually leads to all days scheduled except when used as filter
     */
    None=0,

    /**
     * monday
     */
    Monday=1,

    /**
     * tuesday
     */
    Tuesday=2,

    /**
     * wednesday
     */
    Wednesday=4,

    /**
     * thursday
     */
    Thursday=8,

    /**
     * friday
     */
    Friday=16,

    /**
     * saturday
     */
    Saturday=32,

    /**
     * sunday
     */
    Sunday=64,

    /**
     * all days set
     */
    All=127
}

export namespace ScheduledDays {

    export function getName(type: any): string {
        if(type as string)
        {
            let num=parseInt(type)
            if(num>=0)
            return ScheduledDays[num];
        }
        return type;
    }

    export function getValue(type: any): ScheduledDays {
        if(type as string) {
            let num=parseInt(type);
            if(num>=0)
                return num;

            const literal=type as string;
            if(literal.indexOf(", ")>-1)
            {
                const literalarray=literal.split(", ");
                let value=ScheduledDays.None;
                literalarray.forEach(l=>value|=ScheduledDays[l]);
                return value;
            }
            return ScheduledDays[literal];
        }
        return type;
    }
}